src/commcare_cloud/environment/main.py

```diff
@@ -77,6 +106,40 @@ def get_vault_var(self, var):
     def get_ansible_user_password(self):
         return self.get_vault_variables()['ansible_sudo_pass']
 
+    def record_vault_loaded_event(self, secrets):
+        if (
+            not self.did_send_vault_loaded_event and
+            self.input_argv is not None and
+            secrets.get('DATADOG_API_KEY') and
+            self.public_vars.get('DATADOG_ENABLED')
+        ):
+            self.did_send_vault_loaded_event = True
+            datadog.initialize(
+                api_key=secrets['DATADOG_API_KEY'],
+                app_key=secrets['DATADOG_APP_KEY'],
+            )
+            quoted_args = [shlex_quote(arg) for arg in self.input_argv[1:]]
+            datadog.api.Event.create(
+                title="commcare-cloud vault loaded",
+                text="commcare-cloud {}".format(' '.join(quoted_args)),
+                tags=["environment:{}".format(self.name)],
+            )
+
+    @contextmanager
+    def suppress_vault_loaded_event(self):
+        """Prevent "run event" from being sent to datadog
+
+        This is only effective if `self.get_vault_variables()` has not
+        yet been called outside of this context manager. If it has been
+        called then the event has already been sent and this is a no-op.
+        """
+        sent = self.did_send_vault_loaded_event
```

Comment:
I think in the first commit, this was a good name for the variable, but now it's being used for a connected but distinct meaning—with this suppression functionality, it might be better called `should_send_vault_loaded_event` (with the inverse value).

Reply:
1a642eacd7d314575f47bdea60e359369361726d